### DISCUSSION TITLE
some error when installing liboqs in windows with mingw32

### DISCUSSION THREAD

**Discussion by k-Artin** [open-quantum-safe/liboqs#1370] [Q&A]
hello 
I am getting the following errors when I install liboqs on windows with mingw32.

$ ninja -j 4
[1118/1132] Building C object tests/CMakeFiles/speed_kem.dir/speed_kem.c.obj
FAILED: tests/CMakeFiles/speed_kem.dir/speed_kem.c.obj
C:\msys64\mingw32\bin\cc.exe -DOQS_COMPILE_OPTIONS=\"[-Werror;-Wall;-Wextra;-Wpedantic;-Wstrict-prot
otypes;-Wshadow;-Wformat=2;-Wfloat-equal;-Wwrite-strings;-O3;-fomit-frame-pointer;-fdata-sections;-f
function-sections;-Wl,--gc-sections;-Wno-maybe-uninitialized;-Wbad-function-cast;-Wno-unknown-pragma
s;-Wno-unused-parameter]\" -D__USE_MINGW_ANSI_STDIO=1 -IC:/msys64/home/liboqs/liboqs-main/_build/in
clude -IC:/msys64/home/liboqs/liboqs-main/src -fvisibility=hidden -Werror -Wall -Wextra -Wpedantic
-Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -O3 -fomit-frame-pointer -fdat
a-sections -ffunction-sections -Wl,--gc-sections -Wno-maybe-uninitialized -Wbad-function-cast -Wno-u
nknown-pragmas -Wno-unused-parameter -std=gnu11 -MD -MT tests/CMakeFiles/speed_kem.dir/speed_kem.c.o
bj -MF tests\CMakeFiles\speed_kem.dir\speed_kem.c.obj.d -o tests/CMakeFiles/speed_kem.dir/speed_kem.
c.obj -c C:/msys64/home/liboqs/liboqs-main/tests/speed_kem.c
In file included from C:/msys64/home/liboqs/liboqs-main/tests/speed_kem.c:17:
C:/msys64/home/liboqs/liboqs-main/tests/ds_benchmark.h: In function '_bench_rdtsc':
C:/msys64/home/liboqs/liboqs-main/tests/ds_benchmark.h:125:16: error: implicit declaration of funct
ion '__rdtsc' [-Werror=implicit-function-declaration]
  125 |         **return __rdtsc();**
      |                ^~~~~~~
cc1.exe: all warnings being treated as errors
[1121/1132] Building C object tests/CMakeFiles/speed_sig.dir/speed_sig.c.obj
FAILED: tests/CMakeFiles/speed_sig.dir/speed_sig.c.obj
C:\msys64\mingw32\bin\cc.exe -DOQS_COMPILE_OPTIONS=\"[-Werror;-Wall;-Wextra;-Wpedantic;-Wstrict-prot
otypes;-Wshadow;-Wformat=2;-Wfloat-equal;-Wwrite-strings;-O3;-fomit-frame-pointer;-fdata-sections;-f
function-sections;-Wl,--gc-sections;-Wno-maybe-uninitialized;-Wbad-function-cast;-Wno-unknown-pragma
s;-Wno-unused-parameter]\" -D__USE_MINGW_ANSI_STDIO=1 -IC:/msys64/home/liboqs/liboqs-main/_build/in
clude -IC:/msys64/home/liboqs/liboqs-main/src -fvisibility=hidden -Werror -Wall -Wextra -Wpedantic
-Wstrict-prototypes -Wshadow -Wformat=2 -Wfloat-equal -Wwrite-strings -O3 -fomit-frame-pointer -fdat
a-sections -ffunction-sections -Wl,--gc-sections -Wno-maybe-uninitialized -Wbad-function-cast -Wno-u
nknown-pragmas -Wno-unused-parameter -std=gnu11 -MD -MT tests/CMakeFiles/speed_sig.dir/speed_sig.c.o
bj -MF tests\CMakeFiles\speed_sig.dir\speed_sig.c.obj.d -o tests/CMakeFiles/speed_sig.dir/speed_sig.
c.obj -c C:/msys64/home/liboqs/liboqs-main/tests/speed_sig.c
In file included from C:/msys64/home/liboqs/liboqs-main/tests/speed_sig.c:17:
C:/msys64/home/liboqs/liboqs-main/tests/ds_benchmark.h: In function '_bench_rdtsc':
C:/msys64/home/liboqs/liboqs-main/tests/ds_benchmark.h:125:16: error: implicit declaration of funct
ion '__rdtsc' [-Werror=implicit-function-declaration]
  125 |         **return __rdtsc();**
      |                ^~~~~~~
cc1.exe: all warnings being treated as errors
ninja: build stopped: subcommand failed.

what i see is that the mingw32 can not find the **__rdtsc()** function.
i am searching in the internet and find the library **intrin.h**  which has this function.
i think this library is missing in the **ds_benchmark.h**.
Thank you in advance!

**Comment by baentsch** [comment]
This is a discussion board for general OQS-related items; project & platform-specific issues should be discussed there: Please transfer this to a liboqs issue.

> i think this library is missing in the ds_benchmark.h.

Have you validated that by adding the corresponding #include statement? If so, please consider doing a PR, ideally in combination with a `mingw32` CI test case.

**Comment by k-Artin** [comment]
after adding the library this error will be disappear.
 the discussion transfered to a liboqs issue.
